source/MaterialXGenShader/ColorManagementSystem.h

```diff
@@ -72,6 +78,23 @@ class MX_GENSHADER_API ColorManagementSystem
     /// Returns an implementation for a given transform
     virtual ImplementationPtr getImplementation(const ColorSpaceTransform& transform) const = 0;
 
+    /// For a given transform node and a target port type to connect to, determine the appropriate
+    /// input and output ports. If the transform node's port type does not match the target type
+    /// additional "conversion" nodes will created and the appropriate input and output ports on these 
+    /// nodes will be returned.
+    ///
+    /// Possible configirations include the first two cases where no additional conversion nodes are required, and
+    /// the latter two cases where conversions are required
+    /// 
+    ///  color3 input  -> color3 transform node -> color3 output
+    ///  color4 input  -> color4 transform node -> color3 output
+    ///  color3 input -> color3-to-color4 conversion -> color4 transform node -> color4_to_color3 conversion -> color3 output
+    ///  color4 input -> color4_to_color3 conversion -> color3 transform node -> color3-to-color4 conversion -> color4 output
```

Comment:
Out of curiosity, why do we need to allow having color transform nodes of unmatching type connected? Such auto-conversion handling is not supported elsewhere in MaterialX. But maybe there is a good reason for this?

Reply:
OCIO is always color4 so for now I need to promote / demote as necessary.